render/helpers.go

```diff
@@ -145,10 +145,25 @@ func DatePeriodFormat(s string) string {
 	if _, err := strconv.Atoi(s[:4]); err == nil {
 		// Not just displaying year but month as well
 		if len(s) > 5 {
-			// YYYY[space] = 5 characters
-			postYearIndex := 5
-			s = s[postYearIndex:] + " " + s[:4]
-
+			if len(s) < 14 {
```

Comment:
Wouldn't adding an additional if clause to the existing if statement be simpler?

```
if len == 14 {
    // new functionality
} else if len > 5 {
    // existing code
}
```
This would also cover the case where `len > 15` - with this new code that scenario would return an empty string.